src/transformers/trainer_utils.py

```diff
@@ -411,6 +412,11 @@ def __init__(self, skip_memory_metrics=False):
         if is_torch_cuda_available():
             import torch
 
+            self.torch = torch
+            self.gpu = {}
+        elif is_torch_mps_available():
+            import torch
+
```

Comment:
Have you tried a training on apple silicon Macs?

Reply:
The announcement is one thing. If it works with `transformers` with the changes in this PR is another.

Although the PR is valid on its own, it would be nice to know if it's ready to enable `transformers`' training on such devices.

Reply:
Hi, How do I perform such tests with my Apple silicon Mac?